platform_channels/lib/src/basic_message_channel_demo.dart

```diff
@@ -0,0 +1,100 @@
+///
+/// The widget has a [DropDownButton] which can be used to change the [MessageCodec],
+/// and a [TextField] to enter the text and send the message.
+class BasicMessageChannelDemo extends StatefulWidget {
+  @override
+  _BasicMessageChannelDemoState createState() =>
+      _BasicMessageChannelDemoState();
+}
+
+class _BasicMessageChannelDemoState extends State<BasicMessageChannelDemo> {
+  Future<String> platformReply;
+  List<String> codecs = ['BinaryCodec', 'JSONMessageCodec', 'StringCodec'];
+  String codec = 'StringCodec';
+  TextEditingController controller = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    final textStyle = Theme.of(context).textTheme.headline5;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('BasicMessageChannel Demo'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 16),
+        child: Column(
+          children: [
+            Text(
+              'Select a MessageCodec',
+              style: Theme.of(context).textTheme.headline6,
+            ),
+            SizedBox(height: 16),
+            DropdownButton<String>(
+              value: codec,
+              onChanged: (value) {
+                setState(() {
+                  codec = value;
+                });
+              },
+              items: codecs
+                  .map<DropdownMenuItem<String>>((codec) => DropdownMenuItem(
+                        value: codec,
+                        child: Text(codec),
+                      ))
+                  .toList(),
+            ),
+            Expanded(
+              child: Center(
+                child: FutureBuilder<String>(
+                  future: platformReply,
+                  builder: (context, snapshot) {
+                    if (snapshot.connectionState == ConnectionState.none) {
+                      return Text(
+                        'Send a message to get a Reply',
+                        style: textStyle,
+                      );
+                    } else if (snapshot.hasError) {
+                      return Text(snapshot.error.toString());
+                    } else if (snapshot.connectionState ==
+                        ConnectionState.done) {
+                      return Text(
+                        snapshot.data,
+                        style: textStyle,
+                      );
+                    }
+                    return CircularProgressIndicator();
+                  },
+                ),
+              ),
+            ),
+            TextField(
+              controller: controller,
+              decoration: InputDecoration(
+                suffixIcon: IconButton(
+                  icon: Icon(Icons.send),
+                  onPressed: () {
+                    setState(() {
+                      platformReply = PlatformMessages.sendPlatformMessage(
+                        controller.value.text,
```

Comment:
@RedBrogdon Right now, I'm not checking whether the String is empty or not, and sending it anyways to the platform. Let me know if I should add a validation for empty String or not.